hydra-chain-observer/src/Hydra/ChainObserver/NodeClient.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+
+module Hydra.ChainObserver.NodeClient where
+
+import Hydra.Prelude
+
+import Hydra.Cardano.Api (
+  BlockNo,
+  ChainPoint,
+  NetworkId,
+  SocketPath,
+  Tx,
+  UTxO,
+ )
+import Hydra.Cardano.Api.Prelude (TxId)
+import Hydra.Chain (OnChainTx (..))
+import Hydra.Chain.Direct.Tx (
+  HeadObservation (..),
+  observeHeadTx,
+ )
+import Hydra.Contract (ScriptInfo)
+import Hydra.Ledger.Cardano (adjustUTxO)
+import Hydra.Tx.HeadId (HeadId (..))
+
+type ObserverHandler m = [ChainObservation] -> m ()
+
+data ChainObservation
+  = Tick
+      { point :: ChainPoint
+      , blockNo :: BlockNo
+      }
+  | HeadObservation
+      { point :: ChainPoint
+      , blockNo :: BlockNo
+      , onChainTx :: OnChainTx Tx
+      }
+  deriving stock (Eq, Show, Generic)
+
+instance Arbitrary ChainObservation where
+  arbitrary = genericArbitrary
+
+defaultObserverHandler :: Applicative m => ObserverHandler m
+defaultObserverHandler = const $ pure ()
+
+newtype NodeClient m = NodeClient
+  { follow :: NetworkId -> Maybe ChainPoint -> ObserverHandler m -> m ()
+  }
+
+type ChainObserverLog :: Type
```

Comment:
We have pretty much the same types in the `ChainObserver` so I got a little confused about this. Would it be possible to reuse the types at least or perhaps this is not considered part of these changes here?

Reply:
That type has been moved to here :)